### PR TITLE
Tweaks to docs + wrap long lines + add function index

### PR DESCRIPTION
**Pull request by olivroy**
Just a few minor edits to the docs of deprecated functions. Adding hyperlinks to the new functions. 

Ref: https://roxygen2.r-lib.org/articles/rd-formatting.html?q=link#function-links

Edit: I added a few more tweaks to docs to ensure that long lines are wrapped, links are present. 

Edit2: since this PR mostly makes changes to docs, you may want to take a look at the .Rd diffs.

### CODE REVIEW

**Comment by billdenney** [comment]
@olivroy, I just merged your other PR into `main`.  Can you please reconcile the changes between `main` and this PR?

**Comment by olivroy** [comment]
I will fetch locally to redocument

File: R/janitor.R [hunk 1]
```diff
@@ -1,24 +1,7 @@
...
-#'
-#' janitor was built with beginning-to-intermediate R users in mind
-#' and is optimized for user-friendliness.  Advanced users can do most
-#' things covered here, but they can do it faster with janitor and save
-#' their thinking for more fun tasks.
-#'
+#' Advanced users can do most things covered here, but they can do it
+#' faster with janitor and save their thinking for more fun tasks.
+#' @keywords internal
 "_PACKAGE"
```
Comment (olivroy) [inline R/janitor.R:5]: Note that the difference in .Rd is minimal, because by default `"_PACKAGE"` uses the package description from `DESCRIPTION`

File: R/compare_df_cols.R [hunk 2]
```diff
@@ -241,18 +243,18 @@ compare_df_cols_same <- function(..., bind_method = c("bind_rows", "rbind"), ver
...
-#' lenient: for class comparison purposes, the variable is just a "factor".
+#'   as differences for the purposes of identifying mismatches?
+#'   `strict_description = TRUE` is stricter and factors with different
+#'   levels will be treated as different classes.  `FALSE` is more
+#'   lenient: for class comparison purposes, the variable is just a "factor".
 #' @return A character scalar describing the class(es) of an object where if the
 #'   scalar will match, columns in a data.frame (or similar object) should bind
 #'   together without issue.
-#' @family Data frame type comparison
+#' @family data frame type comparison
```
Comment (olivroy) [inline R/compare_df_cols.R:257]: Using lower-case is better, because it will show up as "Other data frame type comparison" in man (feels more like a sentence than "Other Data frame type comparison" in my opinion

File: R/adorn_title.R [hunk 3]
```diff
@@ -38,12 +55,18 @@ adorn_title <- function(dat, placement = "top", row_name, col_name) {
 
     if (inherits(dat, "tabyl")) {
       if (attr(dat, "tabyl_type") == "one_way") {
-        warning("adorn_title is meant for two-way tabyls, calling it on a one-way tabyl may not yield a meaningful result")
+        warning(c(
```
Comment (billdenney) [inline R/adorn_title.R:58]: I find breaking the message harder to read.  Can you please revert this change?

File: R/adorn_title.R [hunk 4]
```diff
@@ -38,12 +55,18 @@ adorn_title <- function(dat, placement = "top", row_name, col_name) {
...
+          "adorn_title is meant for two-way tabyls, ",
+          "calling it on a one-way tabyl may not yield a meaningful result"
+        ))
       }
     }
     if (missing(col_name)) {
       if (!inherits(dat, "tabyl")) {
-        stop("When input is not a data.frame of class tabyl, a value must be specified for the col_name argument")
+        stop(c(
```
Comment (billdenney) [inline R/adorn_title.R:66]: I find breaking the message harder to read.  Can you please revert this change?

File: R/adorn_totals.R [hunk 5]
```diff
@@ -43,7 +60,11 @@ adorn_totals <- function(dat, where = "row", fill = "-", na.rm = TRUE, name = "T
     }
 
     if (length(cols_to_total) == 0) {
-      stop("at least one targeted column must be of class numeric.  Control target variables with the ... argument. adorn_totals should be called before other adorn_ functions.")
+      stop(
```
Comment (billdenney) [inline R/adorn_totals.R:63]: I find breaking the message harder to read.  Can you please revert this change?

File: R/adorn_totals.R [hunk 6]
```diff
@@ -128,7 +153,10 @@ adorn_totals <- function(dat, where = "row", fill = "-", na.rm = TRUE, name = "T
       if (!1 %in% cols_to_total) { # give users the option to total the first column??  Up to them I guess
         col_totals[1, 1] <- name[1] # replace first column value with name argument
       } else {
-        message("Because the first column was specified to be totaled, it does not contain the label 'Total' (or user-specified name) in the totals row")
+        message(
```
Comment (billdenney) [inline R/adorn_totals.R:156]: I find breaking the message harder to read.  Can you please revert this change?

File: R/compare_df_cols.R (hunk 2 above)
Comment (billdenney) [reply R/compare_df_cols.R:257]: That makes sense to me.

File: R/top_levels.R [hunk 7]
```diff
@@ -3,10 +3,10 @@
 #'
 #' Get a frequency table of a factor variable, grouped into categories by level.
 #'
-#' @param input_vec the factor variable to tabulate.
-#' @param n number of levels to include in top and bottom groups
-#' @param show_na should cases where the variable is NA be shown?
-#' @return a data.frame (actually a `tbl_df`) with the frequencies of the
+#' @param input_vec The factor variable to tabulate.
+#' @param n Number of levels to include in top and bottom groups
+#' @param show_na Should cases where the variable is NA be shown?
```
Comment (billdenney) [inline R/top_levels.R:8]: Please put the NA in ticks (``).

**Review by billdenney — COMMENTED** [review]
This looks good to me, overall.  I just have a few, minor comments.

**Comment by olivroy** [comment]
thanks for the review! I addressed your comments

**Comment by billdenney** [comment]
@sfirke, this looks good to me.  Are you good with merging it?

File: R/janitor.R (hunk 1 above)
Comment (sfirke) [reply R/janitor.R:5]: Nice, thanks for explaining how this works.

File: tests/testthat/test-tabyl.R [hunk 8]
```diff
@@ -309,38 +309,22 @@ test_that("NA levels get moved to the last column in the data.frame, are suppres
     y_with_missing[["NA_"]] %>% untabyl(), # column c remains numeric
     data.frame(c = 10, `1` = 1, `2` = 0, NA_ = 1, check.names = FALSE)
   )
-
   # If no NA in 3rd variable, it doesn't appear in split list
   expect_equal(length(dplyr::starwars %>%
     dplyr::filter(species == "Human") %>%
     tabyl(eye_color, skin_color, gender, show_missing_levels = TRUE)), 2)
 
-  # The starwars data set changed in dplyr v 1.0.0 so have two blocks of tests:
```
Comment (sfirke) [inline tests/testthat/test-tabyl.R:318]: Nice job removing this!

**Review by sfirke — APPROVED** [review]
LGTM

**Comment by sfirke** [comment]
This is great, thanks @olivroy and thanks also @billdenney for the review and questions!
